internal/pkg/service/common/servicectx/servicectx.go

```diff
@@ -79,50 +100,59 @@ func New(opts ...Option) (*Process, error) {
 		c.uniqueID = fmt.Sprintf(`%s-%05d`, hostname, pid)
 	}
 
-	// Create channel used by both the signal handler and service goroutines
-	// to notify the main goroutine when to stop the server.
-	errCh := make(chan error, 1)
-
-	// Create channel used to pass the shutdown context to OnShutdownFn callbacks.
-	shutdownCtxCh := make(chan context.Context, 1)
```

Comment:
These two channels have been replace by the `Process.shutdown` - `chan context.Context`.

See `ShutdownReason` function.

The context carries with him `reason` - shutdown trigger error.

BTW: error value was not used anywhere in the previous implementation, so `errCh` could be deleted away.